pkg/controller/nodeipam/ipam/multinetwork_cloud_cidr_allocator.go

```diff
@@ -82,3 +81,44 @@ func resourceName(name string) string {
 	parts := strings.Split(name, "/")
 	return parts[len(parts)-1]
 }
+
+func (ca *cloudCIDRAllocator) NetworkToNodes(network *networkv1.Network) error {
+	k8sNodesList, err := ca.nodeLister.List(labels.Everything())
+	if err != nil {
+		return fmt.Errorf("error fetching nodes: %v", err)
+	}
+	// Need to reschedule all Nodes, since we do not know which one holds the
+	// new Network.
+	for _, node := range k8sNodesList {
+		if network != nil {
+			// filter out nodes that are not part of network
+			if node.Annotations == nil {
+				// skip node w/o any annotation, not possible for it to have any MN
+				continue
+			}
+			northIntfAnn, ok := node.Annotations[networkv1.NorthInterfacesAnnotationKey]
+			if !ok {
+				// skip node w/o "north-interfaces" annotation, no MN
+				continue
+			}
+			northIntf, err := networkv1.ParseNorthInterfacesAnnotation(northIntfAnn)
+			// if err!=nil means there is some format issue with the annotation, lets
+			// re-generate it for that node
+			if err == nil {
```

Comment:
```
if err != nil {
 klog.Errorf("error parsing annotation %v on node %s", northIntfAnn, node.Name)
}
	for _, ele := range northIntf {
					if ele.Network == network.Name {
						err = ca.AllocateOrOccupyCIDR(node)
                                                  // we need to handle this error
						break
					}
				}

Reply:
ca.AllocateOrOccupyCIDR error is not handled in all the other triggers, and as mentioned above, there is no point of failing, since the trigger is not rerun (unless I am mistaken of that part), see https://github.com/kubernetes/cloud-provider-gcp/blob/master/pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go#L130